usersession/userd/settings_test.go

```diff
@@ -225,6 +229,24 @@ func (s *settingsSuite) TestSetInvalidSetting(c *C) {
 	c.Assert(s.mockXdgSettings.Calls(), IsNil)
 }
 
+func (s *settingsSuite) TestSetInvalidValue(c *C) {
+	err := s.settings.Set("default-web-browser", "foo", ":some-dbus-sender")
+	c.Assert(err, ErrorMatches, `cannot set setting "default-web-browser" to value "foo": value not allowed`)
+	c.Assert(s.mockXdgSettings.Calls(), IsNil)
+}
+
+func (s *settingsSuite) TestSetSubInvalidSetting(c *C) {
+	err := s.settings.SetSub("random-setting", "subprop", "foo.desktop", ":some-dbus-sender")
+	c.Assert(err, ErrorMatches, `cannot use setting "random-setting": not allowed`)
```

Comment:
Why does `SetSub` result in an error message that talks about `use`-ing a setting but `Set` in the test above talks about `set`ting it?

Reply:
Hmm, it's the generic whitelisted check. Perhaps we could add some context or tweak the message in a follow up.